### ISSUE TITLE
[CM11] [iyokan] kernal panic reboot

### ISSUE THREAD

**Issue by pteek** [LegacyXperia/local_manifests#746]
My iyokan had been rebooting a lot in the past 3 days. Most of them were while using hangouts which I fixed by reinstalling hangouts.

This one occurred when I have doing heavy multi tasking. There are the most common one and happen 2-3 times a week when the phone is under heavy load.

logs:

http://pastebin.com/gBKZAnFQ

Ask if you need any other logs.

**Comment by pteek** [comment]
Happened again. There frequency have increased in the past week.

http://pastebin.com/DM4ukXie

**Comment by pteek** [comment]
Happened again.
http://pastebin.com/bNMCsD5D

**Comment by mikeNG** [comment]
Looks like SD card issues to me. Try using the phone without SD or with a different one

**Comment by pteek** [comment]
Do you have these reboots on your link/mounts2sd device?

I am getting 1-2 reboots everyday. The phone is not really useful to me without the extra space from sdext. Usually the wake-lock problem appears after these reboots.

**Comment by mikeNG** [comment]
I don't get any reboots using the latest source, neither did I with the latest nightly. Well I guess it's time for a new build soon to see if anything is improved.

**Comment by pteek** [comment]
I got a reboot on 814 build. So it could be specific to my SD.

**Comment by pcfighter** [comment]
try to check SDcard filesystem (connect via usb cable with mass storage mode) in scandisk or other filesystem check tool

**Comment by pteek** [comment]
There are errors on the sdext partition. I will run fsck.

**Comment by pteek** [comment]
My sdcard has write error problems. Can't reproduce this with other sdcards.

**Comment by mikeNG** [comment]
Is it old? maybe it's starting to die

**Comment by pteek** [comment]
No, it is new. I bought it just months ago. I guess I just had bad luck with this one. The exact error is that CMD25 (write multiple blocks) fails and leads to the crash.

I can reproduce this on mass storage mode on windows and with a external sd card reader.

Offtopic: How can I backup my auto brightness curve? I put a lot of time in customising it. 

Also, is there a way we can get the auto brightness curve from CM10? Or is the system changed too much for that?

**Comment by fabri20** [comment]
me too reboot sometimes, but no everday... i´ve clean install and 816 build with coconut

last_kmsg is here, sorry for my bad english

http://pastebin.com/cuXf2NQd

**Comment by fabri20** [comment]
should make another post for this issue?

**Comment by fabri20** [comment]
Happened again. This time I was playing, performance governor and oc 1.4ghz, but was previously only using whatsapp with interactive governor and oc 1.4ghz

http://pastebin.com/bE2u4543

**Comment by pteek** [comment]
Yours doesn't seem to be related to sd card. Let's wit for mike to comment.

**Comment by pcfighter** [comment]
[  651.635467] MFD is Null

probably it's reason of restarting

**Comment by fabri20** [comment]
could fix it myself? this happened in previous builds

**Comment by 9omorra** [comment]
I am having sdcard related kernel panics / reboots with latest LX cm11 version.
My last_kmsg.log after reboot can be found at http://pastebin.com/K1tkMCHk .
Device: 11-20140907-UNOFFICIAL-LegacyXperia-iyokan
Kernel: 3.4.0-LX+ langes@vbox #1

Unfortunately, I could not compare my logs to the original logs from pteek, as they are no longer online.

Anyways, the relevant part of the logs is at the end of my pastebin. First, there is a spurious mmc0 Interrupt, then a mmc1 timeout, then an error in msmsdcc_hw_reset(), with a kernel panic, finally the reboot. Relevant lines include (many lines omited):
[  136.386840] mmc0: SDIO interrupt recieved without function driver claiming an irq
...
[  137.402221] mmc1: CMD18: Data timeout. DAT0 => 1
...
[  137.404724] mmc1: DMA channel flushed (0x80000004)
[  137.404815] Flush data: 0000c003 4f260078 00000000 00400040 00400040 00000003
[  137.405212] mmcblk0: error -110 transferring data, sector 17094984, nr 64, cmd response 0x900, card status 0xb00
[  137.405548] mmc1: msmsdcc_hw_reset disable regulator: failed: -22
...
[  137.511596] Kernel panic - not syncing: Fatal exception
[  137.511779] Rebooting in 1 seconds..

-EINVAL is -22, so msmsdcc_hw_reset() probably fails because the function msmsdcc_setup_vreg() it calls fails in https://github.com/hellsgod/kernel_msm/blob/e7760c0dc3440e099eeeb0a011a33bdb9ef827bf/drivers/mmc/host/msm_sdcc.c#L2464 (well, whatever the right source is)

I tested the sdcard on a PC and found no problems (full readtest with badblocks + several GB of writes in several 1000 files from multiple tasks).

The issue is reproducible with lildebi and may also affect firefox and other apps. Tell me if there's anything else I could do to help fix this issue.

**Comment by pteek** [comment]
My sdcard had CMD25 data timeout (WRITE_MULTIPLE_BLOCK).

 I was able to reproduce the CMD25 write error on PC as well.

Also, I ran the phone without a sd card for days and there were still reboots and mmc1_detect wakelock, this is confusing. I will do testing in my dev environment whenever I can in the future.

**Comment by pcfighter** [comment]
can you try with another SD Card ?  (just for test)

**Comment by pteek** [comment]
I did, my friend's 8 GB card that was included with his iyokan worked perfectly.

**Comment by 9omorra** [comment]
I downgraded from a 32GB UHS Class 1 microSDHC Card to the original 8GB (Class4?) microSDHC Card and had similar reboots. Yet the log is different: http://pastebin.com/6n9mdzjZ

[ 1814.482299] MFD is Null
[ 1814.483947] Restarting system with command ''.

However, the reason may still be the same. In the 32GB SD log, "host->plat->vreg_data" in msmsdcc_setup_vreg() is probably NULL, in the 8GB SD log "MFD" is NULL. Yet I did not find the string "MFD is Null" in the kernel source code.

**Comment by 9omorra** [comment]
Upgrading to LX to  cm-11-20141006-UNOFFICIAL-LegacyXperia-iyokan.zip did not fix the issue, although the announcement says "Updated MMC kernel drivers". A quick test with the original 8GB microSDHC showed reboots just as before.

**Comment by 9omorra** [comment]
AFAICS the issue is gone with cm-11-20141113-UNOFFICIAL-LegacyXperia-iyokan.zip, or at least it is harder to reproduce. To me, 20141113 is a major improvement. I tested with both the 8GB and the 32GB SD Card from above.

**Comment by pteek** [comment]
That is great to hear.

My issue was reproducible on a card reader also tho. Writes failed consistently.

These are not happening these days so maybe it will work fine. I will test it with L builds.

On December 14, 2014 11:14:24 PM GMT+05:30, 9omorra notifications@github.com wrote:

> AFAICS the issue is gone with
> cm-11-20141113-UNOFFICIAL-LegacyXperia-iyokan.zip, or at least it is
> harder to reproduce. To me, 20141113 is a major improvement. I tested
> with both the 8GB and the 32GB SD Card from above.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/LegacyXperia/local_manifests/issues/746#issuecomment-66921264

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by mikeNG** [comment]
Still an issue?

**Comment by pteek** [comment]
Yes. On all the cards I own(2 8GBs class4, 1 16GB class6), using mounts2sd will lead to these reboots.
